### PR TITLE
[Automation] Fix the Uri format issue 

### PR DESCRIPTION
**Pull request by dvbb**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by dvbb** [comment]
Linked issue: [[BUG] System.UriFormatException thrown when retrieving webhooks form an Azure Automation account #34398](https://github.com/Azure/azure-sdk-for-net/issues/34398)

image
Service will return `"uri":""` or `"uri":null` in all GET, LIST methods, we should change `system.uri` type to `string` type to avoid format issue

File: sdk/automation/Azure.ResourceManager.Automation/tests/SessionRecords/WebhookTests/CreateOrUpdateExistGetGetAllAsync.json [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+        "properties": {
+          "isEnabled": false,
+          "expiryTime": "2023-05-25T06:08:34.0928233\u002B00:00",
+          "runbook": {
+            "name": "runbook7879"
+          },
+          "lastInvokedTime": null,
+          "runOn": null,
+          "parameters": null,
+          "uri": "",
```
Comment (live1206) [inline sdk/automation/Azure.ResourceManager.Automation/tests/SessionRecords/WebhookTests/CreateOrUpdateExistGetGetAllAsync.json:463]: @dvbb Out of curiosity, did you manually add this to reproduce the issue?

File: sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs [hunk 2]
```diff
@@ -19,11 +19,13 @@ public class AutomationManagementTestBase : ManagementRecordedTestBase<Automatio
         protected AutomationManagementTestBase(bool isAsync, RecordedTestMode mode)
         : base(isAsync, mode)
         {
+            JsonPathSanitizers.Add("$..uri");
```
Comment (live1206) [inline sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs:22]: Why does Uri need to be sanitized?

File: sdk/automation/Azure.ResourceManager.Automation/tests/SessionRecords/WebhookTests/CreateOrUpdateExistGetGetAllAsync.json (hunk 1 above)
Comment (dvbb) [reply sdk/automation/Azure.ResourceManager.Automation/tests/SessionRecords/WebhookTests/CreateOrUpdateExistGetGetAllAsync.json:463]: Yes, Service will return `"uri":""` or `"uri":null` in all GET, LIST methods, and SDK side will throw uri format issue when  Service  return `"uri":""`

File: sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs (hunk 2 above)
Comment (dvbb) [reply sdk/automation/Azure.ResourceManager.Automation/tests/AutomationManagementTestBase.cs:22]: This uri contains token info, and the pipeline will throw `Credential Scanner Error CSCAN-AZURE0130`

**Review by live1206 — APPROVED** [review]

**Pull request merged**